lib/puppet/provider/network_snmp/cisco_nexus.rb

```diff
@@ -0,0 +1,48 @@
+require 'cisco_node_utils'
+
+# Implementation for the network_snmp type using the Resource API.
+class Puppet::Provider::NetworkSnmp::CiscoNexus
+  def get(context)
+    @network_snmp ||= Cisco::SnmpServer.new
+
+    current_state = {
+      name:     'default',
+      enable:   (@network_snmp.protocol? || context.device.facts['operatingsystem'] == 'ios_xr') ? true : false,
+      contact:  @network_snmp.contact.empty? ? 'unset' : @network_snmp.contact,
+      location: @network_snmp.location.empty? ? 'unset' : @network_snmp.location,
+    }
+
+    [current_state]
+  end
+
+  def set(context, changes)
+    changes.each do |name, change|
+      is = if context.type.feature?('simple_get_filter')
+             change.key?(:is) ? change[:is] : (get(context, [name]) || []).find { |r| r[:name] == name }
+           else
+             change.key?(:is) ? change[:is] : (get(context) || []).find { |r| r[:name] == name }
+           end
+      context.type.check_schema(is) unless change.key?(:is)
+      should = change[:should]
+      context.updating(name) do
+        update(context, name, should)
+      end
+    end
```

Comment:
This set could probably be just the:

```
      should = change[:should]
      context.updating(name) do
        update(context, name, should)
      end
```

As the likes of the `is` is not required for anything, as it would only be if you need to compare the `is[:ensure]` to see if you need to `update`, `delete` or `create`; but as the type is not `ensurable` then the use of `is` is not required for this type/provider.

Reply:
Good call, copypasta but it's not needed